modules/tinymce/src/themes/silver/main/ts/ui/core/color/ColorSwatch.ts

```diff
@@ -17,21 +18,9 @@ export interface ColorSwatchDialogData {
 
 type ColorFormat = 'forecolor' | 'hilitecolor';
 
-const hasStyleApi = (node: Node): node is (Node & ElementCSSInlineStyle) =>
-  Type.isNonNullable((node as HTMLElement).style);
-
 const getCurrentColor = (editor: Editor, format: ColorFormat): Optional<string> => {
-  let color: string | undefined;
-
-  editor.dom.getParents(editor.selection.getStart(), (elm) => {
-    const value = hasStyleApi(elm) ? elm.style[format === 'forecolor' ? 'color' : 'backgroundColor'] : null;
-
-    if (value) {
-      color = color ? color : value;
-    }
-  });
-
-  return Optional.from(color);
+  const cssRgbValue = Css.get(SugarElement.fromDom(editor.selection.getStart()), format === 'hilitecolor' ? 'background-color' : 'color');
```

Comment:
One side effect of this is that previously it never used the computed style... it only used the raw styles (e.g. CSS stylesheets wouldn't impact this). Are we sure we want it to use the computed style instead here now?

Reply:
This was discussed with husky and decided that we will use the css stylesheets, meaning that the current code is the one we want to go with.

Reply:
I agree, I think the colour the user sees is the one that should be in the dialog.